packages/payload/src/config/orderable/index.ts

```diff
@@ -213,6 +194,52 @@ export const addOrderableEndpoint = (config: SanitizedConfig) => {
         collection.access.update,
       )
     }
+    /**
+     * If there is no target.key, we can assume the user enabled `orderable`
+     * on a collection with existing documents, and that this is the first
+     * time they tried to reorder them. Therefore, we perform a one-time
+     * migration by setting the key value for all documents. We do this
+     * instead of enforcing `required` and `unique` at the database schema
+     * level, so that users don't have to run a migration when they enable
+     * `orderable` on a collection with existing documents.
+     */
+    if (!target.key) {
+      const { docs } = await req.payload.find({
+        collection: collection.slug,
+        req,
+        select: { [orderableFieldName]: true },
+        where: {
+          [orderableFieldName]: {
+            exists: false,
+          },
+        },
+      })
+      for (const doc of docs) {
+        await req.payload.update({
+          id: doc.id,
+          collection: collection.slug,
+          data: {
+            // no data needed since the order hooks will handle this
+          },
+          req,
+        })
+      }
```

Comment:
This can be improved by combining the `update` and `find` args.

```ts
await req.payload.update({
  collection: collection.slug,
  where: {
    [orderableFieldName]: { exists: false }
  },
  data: {},
})
```

Reply:
I had tried. Unfortunately, it doesn't work because all documents would end with the same order key (`a0`). I've added an explanatory comment.

Reply:
Ah, I think I ran into that with the migration too, forgot.